src/GuessTheNumber.java

```diff
@@ -1,21 +1,29 @@
 public class GuessTheNumber {
+    private int tries;
     private final int number;
     private boolean isOver = false;
+    private IOInterface io;
+    int input;
 
     public GuessTheNumber(IOInterface io, int number) {
         this.number = number;
+        this.io = io;
+        this.tries = 1;
     }
 
     public boolean isMatch(int guessed) {
         if (guessed == number) {
-            return true;
+            return isOver = true;
```

Comment:
This looks odd... does this return the result of assigning `true` to `isOver`? Is that any different from just returning `true`?